include/GafferImage/ImagePlug.h

```diff
@@ -166,6 +166,36 @@ class GAFFERIMAGE_API ImagePlug : public Gaffer::ValuePlug
 		};
 		//@}
 
+		/// Utility class for managing accessing plugs in the correct view
+		/// for image nodes with multiple inputs.  validViews should be set to the
+		/// viewNames() of the plug you are accessing.
+		///
+		/// If the current view in the context is valid ( either there is a view, and
+		/// it is in validViews, or validViews is empty and there is no view set ),
+		/// then ViewCheckScope does nothing, and isPresent will return true.
+		///
+		/// If there is a view set, but validViews is empty, then ViewCheckScope will
+		/// create a scope that remove view from the context, and isValid will return
+		/// true ( a viewless input should affect every view ).
```

Comment:
Are we sure this is the right behaviour? Hypothetical example :

- I am merging a viewless image over a stereo one.
- I want to tweak something in that viewless image for one of the destination views.
- If we delete the `image:view` context variable, then I need to use CreateViews to convert my image to stereo first, then tweak it downstream of the CreateViews.
- If we just leave `image:view` in place, I can just go ahead and use it directly in an expression or spreadsheet.

In other words, does it make sense for the default view to be treated as "all views" rather than "no view"?

Reply:
I think what I'm saying here is probably a red herring, and it doesn't sit well with the other stuff I'm saying about having a `default` view. I'll leave it here though, in case it sparks any thoughts...

Reply:
Yeah, I think this is a red herring - I think we may want to treat the default view as "all views" when we use it, but I think we still need to clean up the view context before evaluating plugs ( rather than source nodes needing to accept queries of views which don't exist ).